_ml-commons-plugin/api.md

```diff
@@ -0,0 +1,712 @@
+                        "range": {
+                            "k1": {
+                                "gte": 0
+                            }
+                        }
+                    }
+                ]
+            }
+        },
+        "size": 10
+    },
+    "input_index": [
+        "test_data"
+    ]
+}
+```
+
+### Example: Train and predict with data directly
+
+```json
+POST /_plugins/_ml/_train_predict/kmeans
+{
+    "parameters": {
+        "centroids": 2,
+        "iterations": 1,
+        "distance_type": "EUCLIDEAN"
+    },
+    "input_data": {
+        "column_metas": [
+            {
+                "name": "k1",
+                "column_type": "DOUBLE"
+            },
+            {
+                "name": "k2",
+                "column_type": "DOUBLE"
+            }
+        ],
+        "rows": [
+            {
+                "values": [
+                    {
+                        "column_type": "DOUBLE",
+                        "value": 1.00
+                    },
+                    {
+                        "column_type": "DOUBLE",
+                        "value": 2.00
+                    }
+                ]
+            },
+            {
+                "values": [
+                    {
+                        "column_type": "DOUBLE",
+                        "value": 1.00
+                    },
+                    {
+                        "column_type": "DOUBLE",
+                        "value": 4.00
+                    }
+                ]
+            },
+            {
+                "values": [
+                    {
+                        "column_type": "DOUBLE",
+                        "value": 1.00
+                    },
+                    {
+                        "column_type": "DOUBLE",
+                        "value": 0.00
+                    }
+                ]
+            },
+            {
+                "values": [
+                    {
+                        "column_type": "DOUBLE",
+                        "value": 10.00
+                    },
+                    {
+                        "column_type": "DOUBLE",
+                        "value": 2.00
+                    }
+                ]
+            },
+            {
+                "values": [
+                    {
+                        "column_type": "DOUBLE",
+                        "value": 10.00
+                    },
+                    {
+                        "column_type": "DOUBLE",
+                        "value": 4.00
+                    }
+                ]
+            },
+            {
+                "values": [
+                    {
+                        "column_type": "DOUBLE",
+                        "value": 10.00
+                    },
+                    {
+                        "column_type": "DOUBLE",
+                        "value": 0.00
+                    }
+                ]
+            }
+        ]
+    }
+}
+```
+
+### Response
+
+```json
```

Comment:
Actually the input data should be clustered into two groups. How about we use this response?
```
{
  "status" : "COMPLETED",
  "prediction_result" : {
    "column_metas" : [
      {
        "name" : "ClusterID",
        "column_type" : "INTEGER"
      }
    ],
    "rows" : [
      {
        "values" : [
          {
            "column_type" : "INTEGER",
            "value" : 1
          }
        ]
      },
      {
        "values" : [
          {
            "column_type" : "INTEGER",
            "value" : 1
          }
        ]
      },
      {
        "values" : [
          {
            "column_type" : "INTEGER",
            "value" : 1
          }
        ]
      },
      {
        "values" : [
          {
            "column_type" : "INTEGER",
            "value" : 0
          }
        ]
      },
      {
        "values" : [
          {
            "column_type" : "INTEGER",
            "value" : 0
          }
        ]
      },
      {
        "values" : [
          {
            "column_type" : "INTEGER",
            "value" : 0
          }
        ]
      }
    ]
  }
}

```